defaults/jq/semconv.jq

```diff
@@ -104,3 +108,51 @@ def semconv_grouped_metrics($options): semconv_metrics($options) | semconv_group
 
 # Convenience function to group all metrics by their root namespace without any filtering options.
 def semconv_grouped_metrics: semconv_grouped_metrics({});
+
+#################
+# Event Functions
+#################
+
+# Extracts and processes semantic convention events based on provided options.
+# $options is an object that can contain:
+# - exclude_stability: a list of stability statuses to exclude.
+# - exclude_deprecated: a boolean to exclude deprecated events.
+# - exclude_event_namespace: a list of event domain to exclude.
+def semconv_events($options):
+    .groups
+    | map(select(.type == "event"))
+    | if ($options | has("exclude_stability")) then
+        map(select(.stability as $st | $options.exclude_stability | index($st) | not))
+      else
+        .
+      end
+    | if ($options | has("exclude_deprecated") and $options.exclude_deprecated == true) then
+        map(select(.id | endswith(".deprecated") | not))
```

Comment:
Thats exactly what this is doing?? -- marking the `event` (the whole group) as deprecated?

Reply:
it should not use group id - there is a `deprecated`  property on the event group which should be used instead.


I.e.
```yaml
  - id: az.resource.log
    type: event
    deprecated: "foobar"
    name: az.resource.log
    ...
```

and then the event-specific JQ filters should use the explicit property instead of unreliable group id.

Reply:
All changes in this file have been reverted